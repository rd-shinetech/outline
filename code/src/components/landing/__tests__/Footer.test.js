import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from '../Footer';


describe('Footer.test.js', () => {

    it('renders without crashing', () => {
        
        const wrapper = shallow(<Footer />);

        expect(wrapper).toHaveLength(1);

        expect(wrapper.find('footer')).toHaveLength(1);
        
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
        
    });
});
