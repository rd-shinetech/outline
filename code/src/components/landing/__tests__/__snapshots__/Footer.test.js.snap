exports[`Footer.test.js renders without crashing 1`] = `
<footer>
  <h2
    className="display-4 text-center py-5 my-4">
    Features
  </h2>
  <nav
    className="nav justify-content-center nav-pills flex-column flex-md-row">
    <a
      className="nav-link active"
      data-toggle="tab"
      href="#rest">
      Mock REST
    </a>
    <a
      className="nav-link"
      data-toggle="tab"
      href="#unitTesting">
      Unit Tests
    </a>
    <a
      className="nav-link"
      data-toggle="tab"
      href="#es2015">
      ES2015
    </a>
    <a
      className="nav-link"
      data-toggle="tab"
      href="#others">
      Others
    </a>
  </nav>
  <div
    className="tab-content py-5">
    <div
      className="tab-pane active"
      id="rest">
      <h3>
        Mock REST
      </h3>
      <p>
        Custom written mock REST
      </p>
    </div>
    <div
      className="tab-pane"
      id="es2015">
      <h3>
        ES2015
      </h3>
      <p>
        aka ES6
      </p>
    </div>
    <div
      className="tab-pane"
      id="unitTesting">
      <h3>
        Unit Testing
      </h3>
      <ul>
        <li>
          Jest
        </li>
        <li>
          Enzyme
        </li>
        <li>
          Nock
        </li>
        <li>
          Expect assertion
        </li>
        <li>
          Code coverage
        </li>
      </ul>
    </div>
    <div
      className="tab-pane"
      id="others">
      <h3>
        Others
      </h3>
      <ul>
        <li>
          Redux
        </li>
        <li>
          Redux Form
        </li>
        <li>
          Lodash
        </li>
        <li>
          React Bootstrap Table
        </li>
        <li>
          Font Awesome (for icons)
        </li>
        <li>
          Hot Module Replacement (HMR)
        </li>
      </ul>
    </div>
  </div>
</footer>
`;
